src/settings/controllers/ServerSupportUnstableFeatureController.ts

```diff
@@ -26,12 +26,21 @@ import SettingsStore from "../SettingsStore";
  * When a setting gets disabled or enabled from this controller it notifies the given WatchManager
  */
 export default class ServerSupportUnstableFeatureController extends MatrixClientBackedController {
+    // Starts off as `undefined` so when we first compare the `newDisabledValue`, it sees
+    // it as a change and updates the watchers.
     private enabled: boolean | undefined;
 
+    /**
+     * Construct a new ServerSupportUnstableFeatureController.
+     *
+     * @param unstableFeatureGroups - If any one of the feature groups is satisfied,
+     * then the setting is considered enabled. A feature group is satisfied if all of
+     * the features in the group are supported (all features in a group are required).
+     */
```

Comment:
*As suggested by @t3chguy in https://github.com/matrix-org/matrix-react-sdk/pull/10398#discussion_r1139471244,*

> Guess it should be extended to string[][] where its an OR of AND groups to offer more flexibility

Refactoring `ServerSupportUnstableFeatureController` to support multiple feature groups where any one of them will enable the setting. All features in a feature group are required.

This way having either `org.matrix.msc3030` or `org.matrix.msc3030.stable` will enable the jump to date feature flag with a config of `[["org.matrix.msc3030"], ["org.matrix.msc3030.stable"]]`

If you want to *AND* feature together like before, just add them to the same group: `[["foo", "bar"]]`